app/scripts/lib/rpc-method-middleware/handlers/add-ethereum-chain.js

```diff
@@ -5,23 +5,22 @@ import {
   MESSAGE_TYPE,
```

Comment:
Are there tests for `addEthereumChain` and `switchEthereumChain`? If these are being tested indirectly through other tests no worries. Or have you tested this with the test dapp? Just want to make sure these changes don't break anything as they seem important.

Reply:
They're both covered in some e2e tests. Not super comprehensively though. Fair point that we should have unit tests for these handlers

Reply:
yeah so unfortunately I don't believe we do test these. Feels like we should. Not exactly sure what the obstacles are? I think the path there would involve removing the localhost network from preloads in the e2e test build and then adding a button on the test-dapp to add it via `wallet_addEthereumChain` RPC API. Will speak to dapp api team about this. But for now I've been testing manually like this:

https://user-images.githubusercontent.com/34557516/224126760-8898c9e2-dad7-4e5d-915a-1168dfe25e00.mov

Reply:
See `test/e2e/tests/chain-interactions.spec.js` and `add-custom-network.spec.js` for some examples of these being covered in e2e tests

Reply:
Oh! I didn't know about the `chain-interactions` e2e test! yeah that covers it!

Reply:
Some of the tests use injected JS to trigger events; we don't even really need the button. But yeah good catch, I don't see `wallet_switchEthereumChain` being invoked either.

Reply:
Tracked here: https://github.com/MetaMask/metamask-extension/issues/18088

And unit tests are tracked here: https://github.com/MetaMask/metamask-extension/issues/18089